### PR TITLE
Fix issue where icon on RadioMenuFlyout and ToggleMenuFlyout was not being displayed

### LINKED ISSUE #4473: ToggleMenuFlyoutItem and RadioMenuFlyoutItem do not show assigned icon in winui 2.6 prerelease
**Describe the bug**

When using the new styles for both controls, the assigned icon is ignored.

**Steps to reproduce the bug**

1. Create a new winui 2.6 app and change the styles version to "latest"
2. Add a ToggleMenuFlyoutItem or RadioMenuFlyoutItem
3. Assign an icon to the item 
4. Observe that the icon is not shown

**Expected behavior**

Icon should be shown

**Screenshots**

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml.2.6.0-prerelease.210227001]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes  |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Update templates to contain the icon viewbox and update visual states to update the state correctly.

## Motivation and Context

Closes #4473 
## How Has This Been Tested?

Tested manually.

### CODE REVIEW

**Comment by ranjeshj** [comment]
@chingucoding can you share pics of what the states look like now. Also, can you take a quick look at the original PR and see if we are missing any other template parts ? Thanks!

**Comment by chingucoding** [comment]
> @chingucoding can you share pics of what the states look like now. Also, can you take a quick look at the original PR and see if we are missing any other template parts ? Thanks!

Screenshots:

Image of ToggleMenuFlyoutItem with icon being set

Image of RadioMenuFlyoutItem with icon being set

The only thing missing in the templates are the "KeyboardAcceleratorTextBlock", however I am not sure if this was a deliberate choice removing them or not. Maybe @tashatitova knows more about this?

**Comment by ranjeshj** [comment]
> > @chingucoding can you share pics of what the states look like now. Also, can you take a quick look at the original PR and see if we are missing any other template parts ? Thanks!
> 
> Screenshots:
> 
> Image of ToggleMenuFlyoutItem with icon being set
> 
> Image of RadioMenuFlyoutItem with icon being set
> 
> The only thing missing in the templates are the "KeyboardAcceleratorTextBlock", however I am not sure if this was a deliberate choice removing them or not. Maybe @tashatitova knows more about this?

I don't think that is deliberate. We should add that back.

**Comment by chingucoding** [comment]
Readded them now @ranjeshj .

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 1]
```diff
@@ -480,6 +480,10 @@
                                            Storyboard.TargetProperty="Foreground">
                                             <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource MenuFlyoutItemForegroundPointerOver}" />
                                         </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="IconContent"
+                                           Storyboard.TargetProperty="Foreground">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource MenuFlyoutItemForegroundPointerOver}" />
+                                        </ObjectAnimationUsingKeyFrames>
```
Comment (StephenLPeters) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:486]: The reveal style for toggle menu flyout item also sets the accellerator key textbox foreground, are we missing that?  
                                        <contract6Present:Setter Target="KeyboardAcceleratorTextBlock.Foreground" Value="{ThemeResource ToggleMenuFlyoutItemKeyboardAcceleratorTextForegroundDisabled}" />

File: dev/CommonStyles/TestUI/MenuFlyoutPage.xaml [hunk 2]
```diff
@@ -30,24 +30,25 @@
...
+            <Button.Flyout>
+                <MenuFlyout>
+                    <MenuFlyoutItem Text="Share">
+                        <MenuFlyoutItem.Icon>
+                            <FontIcon Glyph="&#xE72D;"/>
+                        </MenuFlyoutItem.Icon>
+                    </MenuFlyoutItem>
+                    <MenuFlyoutItem Text="Copy" Icon="Copy"/>
+                    <MenuFlyoutItem Text="Delete" Icon="Delete"/>
+                    <ToggleMenuFlyoutItem Text="ToggleMenuFlyoutItem" Icon="Delete" />
```
Comment (StephenLPeters) [inline dev/CommonStyles/TestUI/MenuFlyoutPage.xaml:43]: Can we add a RadioMenuFLyoutItem as well?

Comment (chingucoding) [reply dev/CommonStyles/TestUI/MenuFlyoutPage.xaml:43]: We can do that, but there already is an entire page dedicated to that control. But if you still want to, I can add it.

**Comment by ranjeshj** [comment]
@chingucoding can you share pics once more with the possible combinations ? @YuliKl @marksfoster  can you double check this is good ?

**Comment by chingucoding** [comment]
Sure Ranjesh, here more elaborate screenshots:

Screenshot of multiple different flyout items with icons and keyboard accelerators set

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 3]
```diff
@@ -508,6 +516,10 @@
                                                 Storyboard.TargetProperty="Foreground">
                                             <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource MenuFlyoutSubItemForegroundDisabled}" />
                                         </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="Icon"
```
Comment (teaP) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:519]: I think this should be IconContent also?

**Review by teaP — CHANGES_REQUESTED** [review]
When I look at the generic.xaml version of the styles for MenuFlyoutItem and ToggleMenuFlyoutItem, they look much more up to date than what we currently have in here (e.g. they have Setters instead of these archaic Storyboards, the keyboard accelerator is in there, etc.). I'm not sure what the history is here, but it looks to me like both of these styles actually need to be updated to what generic.xaml has, minus reveal. Is that something you'd want to take on?

**Comment by chingucoding** [comment]
#3958 deliberately changed the setters to objectanimations, that's why I thought that this would be the new way moving forward for these styles. So what is the correct way of doing this now? Were the changes of #3958 "incorrect" then @teaP ?

File: dev/CommonStyles/MenuFlyout_themeresources.xaml (hunk 1 above)
Comment (chingucoding) [reply dev/CommonStyles/MenuFlyout_themeresources.xaml:489]: Good catch, thank you!

File: dev/CommonStyles/MenuFlyout_themeresources.xaml (hunk 3 above)
Comment (chingucoding) [reply dev/CommonStyles/MenuFlyout_themeresources.xaml]: Good catch, that was a typo.

**Comment by teaP** [comment]
> #3958 deliberately changed the setters to objectanimations, that's why I thought that this would be the new way moving forward for these styles. So what is the correct way of doing this now? Were the changes of #3958 "incorrect" then @teaP ?
> 
> Edit: Looks like that there only was a movement of code, not replacement. None the less, what is the correct way here then?

We definitely should stick with setters anywhere there's no actual animation. I should have looked at #3958 more closely :P especially since it looks like we lost some things that you're adding back now.

**Comment by chingucoding** [comment]
> We definitely should stick with setters anywhere there's no actual animation. I should have looked at #3958 more closely :P especially since it looks like we lost some things that you're adding back now.

Well updated the menu flyout item styles now holistically to use setters where I was confident we can use a setter instead!

Out of curiosity, @teaP do you know why animations were used in the first place?

**Comment by teaP** [comment]
> > We definitely should stick with setters anywhere there's no actual animation. I should have looked at #3958 more closely :P especially since it looks like we lost some things that you're adding back now.
> 
> Well updated the menu flyout item styles now holistically to use setters where I was confident we can use a setter instead!
> 
> Out of curiosity, @teaP do you know why animations were used in the first place (as in why were those chosen few years ago instead of setters)?

Thanks! Animations were historically used because VisualState.Setters didn't used to exist. :) It was added in TH1.

**Comment by chingucoding** [comment]
> Thanks! Animations were historically used because VisualState.Setters didn't used to exist. :) It was added in TH1.

OHhh I see, that makes sense, thank you @teaP !

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 4]
```diff
@@ -467,48 +452,28 @@
...
-                                           Storyboard.TargetProperty="Foreground">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource MenuFlyoutSubItemForegroundPressed}" />
-                                        </ObjectAnimationUsingKeyFrames>
-                                    </Storyboard>
+                                    <VisualState.Setters>
+                                        <Setter Target="LayoutRoot.Background" Value="{ThemeResource MenuFlyoutSubItemBackgroundPressed}"/>
+                                        <Setter Target="TextBlock.Foreground" Value="{ThemeResource MenuFlyoutSubItemForegroundPressed}"/>
+                                        <Setter Target="CheckGlyph.Foreground" Value="{ThemeResource MenuFlyoutSubItemForegroundPressed}"/>
+                                        <Setter Target="IconContent.Foreground" Value="{ThemeResource MenuFlyoutSubItemForegroundPressed}"/>
+                                    </VisualState.Setters>
```
Comment (teaP) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:469]: Missing the keyboard accelerator foreground here (and also below in the Disabled state).

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 5]
```diff
@@ -521,6 +486,24 @@
                                             Duration="0" />
```
Comment (teaP) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:486]: Switch this to 
```
<VisualState.Setters>
    <Setter Target="CheckGlyph.Opacity" Value="1" />
</VisualState.Setters>
```

?

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 6]
```diff
@@ -521,6 +486,24 @@
...
+                                <VisualState x:Name="CheckedWithIcon">
+                                    <VisualState.Setters>
+                                        <Setter Target="TextBlock.Margin" Value="{ThemeResource MenuFlyoutItemPlaceholderThemeThickness}" />
+                                        <Setter Target="IconRoot.Visibility" Value="Visible" />
+                                    </VisualState.Setters>
+                                    <Storyboard>
+                                        <DoubleAnimation Storyboard.TargetName="CheckGlyph"
+                                            Storyboard.TargetProperty="Opacity"
+                                            To="1"
+                                            Duration="0" />
```
Comment (teaP) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:504]: Switch this to being a setter

File: dev/CommonStyles/MenuFlyout_themeresources.xaml [hunk 7]
```diff
@@ -588,62 +601,34 @@
...
-                                        </ObjectAnimationUsingKeyFrames>
-                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SubItemChevron" Storyboard.TargetProperty="Foreground">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource MenuFlyoutSubItemChevronPointerOver}" />
-                                        </ObjectAnimationUsingKeyFrames>
-                                    </Storyboard>
+                                    <VisualState.Setters>
+                                        <Setter Target="LayoutRoot.Background" Value="{ThemeResource MenuFlyoutSubItemBackgroundPointerOver}"/>
+                                        <Setter Target="TextBlock.Foreground" Value="{ThemeResource MenuFlyoutSubItemForegroundPointerOver}"/>
+                                        <Setter Target="SubItemChevron.Foreground" Value="{ThemeResource MenuFlyoutSubItemChevronPointerOver}"/>
+                                    </VisualState.Setters>
```
Comment (teaP) [inline dev/CommonStyles/MenuFlyout_themeresources.xaml:608]: Can you please also add back the setters for IconContent.Foreground here too? (And below for Pressed and Disabled as well.)

**Review by teaP — CHANGES_REQUESTED** [review]
Thanks for helping to clean this up!

**Comment by chingucoding** [comment]
Problems should be resolved now.

**Review by teaP — APPROVED** [review]

**Pull request merged**